lib/src/rules/annotate_overrides.dart

```diff
@@ -116,10 +125,17 @@ class _Visitor extends SimpleAstVisitor<void> {
     }
   }
 
+
   @override
   void visitMethodDeclaration(MethodDeclaration node) {
     if (node.isStatic) return;
 
     check(node.declaredElement, node.name);
   }
+
+  @override
+  void visitMixinDeclaration(MixinDeclaration node) {
+    // Clear the (possibly) cached element.
+    classElement = null;
```

Comment:
The declared element of a mixin is a `ClassElement`, so you could capture it here. And mixins can override members from the `on` class, so it's probably the right thing to do.

And given that no tests broke as a result of this change, we should add some tests of these cases.